source/base/mpi.cc

```diff
@@ -181,6 +182,29 @@ namespace Utilities
 
 
 
+    std::vector<IndexSet>
+    create_ascending_partitioning(const MPI_Comm &           comm,
+                                  const IndexSet::size_type &local_size)
+    {
+      const unsigned int n_proc = n_mpi_processes(comm);
+      const auto         sizes  = all_gather(comm, local_size);
```

Comment:
I had to look up what `all_gather` returns. I think it would be easier to read if you spelled out the actual type of `sizes` heer.